.github/workflows/canary-integration-test.yml

```diff
@@ -884,4 +884,177 @@ jobs:
 
     - name: setup tmate session for debugging
       if: failure() && contains(github.event.pull_request.labels.*.name, 'debug-lvm-pvc')
-      uses: mxschmitt/action-tmate@v3
\ No newline at end of file
+      uses: mxschmitt/action-tmate@v3
+
+  multi-cluster-mirroring:
+    runs-on: ubuntu-18.04
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+
+    - name: setup golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+
+    - name: install deps
+      run: |
+        sudo wget https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64 -O /usr/bin/yq
+        sudo chmod +x /usr/bin/yq
+
+    - name: setup minikube
+      uses: manusa/actions-setup-minikube@v2.3.1
+      with:
+        minikube version: 'v1.18.1'
+        kubernetes version: 'v1.19.2'
+        start args: --memory 6g --cpus=2
+        github token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: use local disk into two partitions
+      run: |
+        BLOCK=$(sudo lsblk --paths|awk '/14G/ {print $1}'| head -1)
+        BLOCK_DATA_PART=${BLOCK}1
+        sudo dmsetup version||true
+        sudo swapoff --all --verbose
+        sudo umount /mnt
+        # search for the device since it keeps changing between sda and sdb
+        sudo wipefs --all --force "$BLOCK_DATA_PART"
+        tests/scripts/create-bluestore-partitions.sh --disk "$BLOCK" --osd-count 2
+        sudo lsblk
+
+    - name: build rook
+      run: |
+        # set VERSION to a dummy value since Jenkins normally sets it for us. Do this to make Helm happy and not fail with "Error: Invalid Semantic Version"
+        GOPATH=$(go env GOPATH) make clean && make -j$nproc IMAGES='ceph' VERSION=0 build
+        docker images
+        docker tag $(docker images|awk '/build-/ {print $1}') rook/ceph:master
+
+    - name: deploy first cluster rook
+      run: |
+        BLOCK=$(sudo lsblk|awk '/14G/ {print $1}'| head -1)
+        cd cluster/examples/kubernetes/ceph/
+        kubectl create -f crds.yaml -f common.yaml -f operator.yaml
+        yq w -i -d1 cluster-test.yaml spec.dashboard.enabled false
+        yq w -i -d1 cluster-test.yaml spec.storage.useAllDevices false
+        yq w -i -d1 cluster-test.yaml spec.storage.deviceFilter ${BLOCK}1
+        kubectl create -f cluster-test.yaml -f rbdmirror.yaml -f filesystem-mirror.yaml -f toolbox.yaml
+
+    - name: deploy second cluster rook
+      run: |
+        BLOCK=$(sudo lsblk|awk '/14G/ {print $1}'| head -1)
+        cd cluster/examples/kubernetes/ceph/
+        sed -i 's/namespace: rook-ceph/namespace: rook-ceph-external/g' cluster-test.yaml rbdmirror.yaml filesystem-mirror.yaml
+        kubectl create -f common-external.yaml
```

Comment:
We're not creating an external cluster, we're creating a second cluster, right? Instead of using the common-external.yaml, then creating service accounts and RBAC specifically in this test, can we find a way to extract what we need from common.yaml? This would really benefit upstream users who also want to create the second cluster. 

For example, maybe we put a marker in common.yaml and create everything after that marker in the second namespace. Things like ClusterRoles that are cluster-wide would be above the marker. Maybe there is a better way to implement it, but what do you think?

Reply:
Honestly, I was trying not to create yet another YAML, so I used `common-external.yaml`, even though it's not accurate, but who cares, it's the CI.

Using more markers in `common.yaml` feels an unnecessary extra effort. I'm leaning towards just using a YAML file from the `tests/manifests` directory.